lib/ddtrace/span.rb

```diff
@@ -60,26 +58,22 @@ class Span
 
     attr_accessor :name, :service, :span_type,
                   :span_id, :trace_id, :parent_id,
-                  :status, :sampled,
-                  :tracer, :context
+                  :status, :sampled
 
     attr_reader :parent, :start_time, :end_time, :resource_container
 
     attr_writer :duration
 
-    # Create a new span linked to the given tracer. Call the \Tracer method <tt>start_span()</tt>
-    # and then <tt>finish()</tt> once the tracer operation is over.
+    # Create a new span manually. Call the <tt>start()</tt> method to start the time
+    # measurement and then <tt>stop()</tt> once the timing operation is over.
     #
     # * +service+: the service name for this span
     # * +resource+: the resource this span refers, or +name+ if it's missing.
     #     +nil+ can be used as a placeholder, when the resource value is not yet known at +#initialize+ time.
     # * +span_type+: the type of the span (such as +http+, +db+ and so on)
     # * +parent_id+: the identifier of the parent span
     # * +trace_id+: the identifier of the root span for this trace
-    # * +context+: the context of the span
-    def initialize(tracer, name, options = {})
-      @tracer = tracer
-
+    def initialize(name, options = {})
```

Comment:
Could do that, if it doesn't cause any problems for the older version of Ruby (2.1+) we still support.

Reply:
It only affects arguments in which you have to tell the difference between the caller providing `nil` vs completely omitting that argument. For example: for `def initialize(opt1: "my-default")`, if you need to the the difference between `Span.new(name, opt1: nil)` and `Span.new(name)`, you'll need the `options = {}` (or `**options`) generic keyword argument capture, instead of a simple keyword argument with default value `opt1: "my-default"`.

This doesn't seem to be the case with any of the arguments for `Span`, so it looks safe to refactor.

Reply:
Indeed there's a few semantic differences like Marco stated, but otherwise I don't think there's any big changes from 2.1 to 2.6 in terms of keyword args that we'd need to be on the look out for (in 2.7 and 3.0 there's the whole separation business).

Reply:
Started to employ this change, but this is not simple. Causes a lot of code to break and chain to other parts of code (`Span` --> `SpanOperation` --> `Tracer`) because `options` are passed around quite freely, and sometimes include unexpected, but harmless args.

While I am for changing to keywords, I think this is definitely a breaking change, and we probably shouldn't do this in this PR. Maybe 1.0.